lib/galaxy/jobs/__init__.py

```diff
@@ -902,10 +902,16 @@ def get_special( ):
     def _create_working_directory( self ):
         job = self.get_job()
         try:
+            # TEMP BLOCK --- START
+            pluggedMedia = None
+            for pM in job.user.pluggedMedia:
```

Comment:
This ``for`` loop just to set a variable is a bit awkward. I think it'd be better to add a method to the ``User`` class in ``lib/galaxy/model/__init__.py`` that defines this.

```
def current_plugged_media(self):
    return None if not len(self.plugged_media) else self.plugged_media[-1]
```

and then access like:

```
job.user.get_current_plugged_media()
```

It is really unfortunate that ``plugged_media`` is also used as both a list and a single instance. I'd consider a name that has a more obvious, distinct singular vs. plural modality (e.g. ``plugged_volume`` vs. ``plugged_volumes``) or use ``plugged_medium`` for the single instance you are fetching.